src/xmpp/muc.c

```diff
@@ -858,6 +864,54 @@ muc_anonymity_type(const char *const room)
     }
 }
 
+/*
+ * Return the list of jid affiliated as member in the room
+ */
+GList*
+muc_members(const char *const room)
+{
+    ChatRoom *chat_room = g_hash_table_lookup(rooms, room);
+    if (chat_room) {
+        return g_hash_table_get_keys(chat_room->members);
+    } else {
+        return NULL;
+    }
+}
+
+void
+muc_members_add(const char *const room, const char *const jid)
+{
+    ChatRoom *chat_room = g_hash_table_lookup(rooms, room);
+    if (chat_room) {
+        if (g_hash_table_insert(chat_room->members, strdup(jid), NULL)) {
+#ifdef HAVE_OMEMO
+            Jid *our_jid = jid_create(connection_get_fulljid());
+            if (strcmp(jid, our_jid->barejid) != 0) {
+                omemo_start_session(jid);
+            }
+#endif
+        }
+    }
+}
+
+void
+muc_members_remove(const char *const room, const char *const jid)
+{
+    ChatRoom *chat_room = g_hash_table_lookup(rooms, room);
+    if (chat_room) {
+        g_hash_table_remove(chat_room->members, strdup(jid));
```

Comment:
How does g_hash_table_remove() behave if the jid is not in the list?
Also why strdup() ?

Reply:
We should remove this spurious strdup. Thanks.